grace/models/classifier.py

```diff
@@ -85,17 +82,39 @@ def __init__(
                     for i in range(len(hidden_channels_list) - 1)
                 ]
             )
-            self.node_classifier = Linear(
-                hidden_channels_list[-1], node_output_classes
-            )
-            self.edge_classifier = Linear(
-                hidden_channels_list[-1] * 2, edge_output_classes
+
+        # Consider more than just one Linear layer to squish output:
+        if not hidden_dense_channels:
+            self.node_dense_list = None
+        else:
+            hidden_channels_list.extend(hidden_dense_channels)
+            self.node_dense_list = ModuleList(
+                [
+                    Linear(
+                        hidden_channels_list[
+                            -len(hidden_dense_channels) + i - 1
+                        ],
+                        hidden_channels_list[-len(hidden_dense_channels) + i],
+                    )
+                    for i in range(len(hidden_dense_channels))
+                ]
             )
+        # Final node & edge classifier:
+        self.node_classifier = Linear(
+            hidden_channels_list[-1], node_output_classes
+        )
+        self.edge_classifier = Linear(
+            hidden_channels_list[-1] * 2, edge_output_classes
+        )
 
         # Don't forget the dropout:
-        # TODO: Implement dropout at other positions, not just before Linear
         self.dropout = dropout
 
+        print(self.conv_layer_list)
```

Comment:
shouldn't we use the logger instead of print?

Reply:
Absolutely, I'll also add the `verbose` option to the model ;)